pipeline/frontend/yaml/compiler/convert.go

```diff
@@ -102,6 +103,15 @@ func (c *Compiler) createProcess(container *yaml_types.Container, stepType backe
 		}
 	}
 
+	for _, requested := range container.Secrets.Secrets {
+		secret, ok := c.secrets[strings.ToLower(requested.Source)]
+		if ok && secret.Available(container) {
+			environment[strings.ToUpper(requested.Target)] = secret.Value
+		} else {
+			return nil, fmt.Errorf("secret %q not found or not allowed to be used", requested.Source)
```

Comment:
If a secret is requested as part of the `secrets` list property of a step we check here.